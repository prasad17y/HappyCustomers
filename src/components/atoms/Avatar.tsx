import React from 'react';
import {View, Text, StyleSheet} from 'react-native';

interface AvatarProps {
  name: string;
}

const Avatar: React.FC<AvatarProps> = ({name}) => {
  return (
    <View style={styles.initialsContainer}>
      <Text style={styles.initialsText}>{getInitials(name)}</Text>
    </View>
  );
};

const getInitials = (name: string) => {
  const names = name.split(' ');
  const firstNameInitial = names[0] ? names[0][0] : '';
  return `${firstNameInitial}`.toUpperCase();
};

const styles = StyleSheet.create({
  initialsContainer: {
    width: 40,
    height: 40,
    borderRadius: 4,
    backgroundColor: '#e3effa',
    justifyContent: 'center',
    alignItems: 'center',
  },
  initialsText: {
    color: '#0b5ac2',
    fontSize: 16,
    fontWeight: 'bold',
  },
});

export default React.memo(Avatar);
