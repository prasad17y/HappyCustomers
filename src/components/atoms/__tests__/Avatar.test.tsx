import React from 'react';
import {render, screen} from '@testing-library/react-native';
import Avatar from '../Avatar';

describe('Avatar', () => {
  it('should render the correct initial from a full name', () => {
    render(<Avatar name="John Doe" />);
    // Check that the component correctly displays the first initial
    expect(screen.getByText('J')).toBeTruthy();
  });

  it('should render correctly and match the snapshot', () => {
    const tree = render(<Avatar name="Jane Doe" />);
    // Create a snapshot to lock in the component's structure and style
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should handle an empty name without crashing and match snapshot', () => {
    const tree = render(<Avatar name="" />);
    // Check that an empty name doesn't cause an error and renders consistently
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
