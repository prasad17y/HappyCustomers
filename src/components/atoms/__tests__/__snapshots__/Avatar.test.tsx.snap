// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should handle an empty name without crashing and match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#e3effa",
      "borderRadius": 4,
      "height": 40,
      "justifyContent": "center",
      "width": 40,
    }
  }
>
  <Text
    style={
      {
        "color": "#0b5ac2",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  />
</View>
`;

exports[`Avatar should render correctly and match the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#e3effa",
      "borderRadius": 4,
      "height": 40,
      "justifyContent": "center",
      "width": 40,
    }
  }
>
  <Text
    style={
      {
        "color": "#0b5ac2",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    J
  </Text>
</View>
`;
